From add79aa6e52b2f12f8370b858aa97685ca52403f Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Mon, 8 Apr 2013 16:35:58 +0200
Subject: [PATCH 8/8] [REVERT ME][AUDIO] Add a setParameter from AudioSystem to
 the Policy

BZ: 217971

Orig-Bug: 163953

Some services use setParameter of AudioSystem to inform
of platform changes. This platform changes cannot be sent to the
policy. FM Rx playback requires for example the policy to be aware of the
requested output device and must inform the policy.

This patch adds a setParameter link from AudioSystem to the policy
to handle generic setParameter calls in the policy if needed.

Change-Id: Ifb6ed2bbae0c38feea5589e4a3e45a1a002d797c
Category: device enablement
Domain: AudioComms-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 include/hardware/audio_policy.h | 3 +++
 modules/audio/audio_policy.c    | 7 +++++++
 2 files changed, 10 insertions(+)

diff --git a/include/hardware/audio_policy.h b/include/hardware/audio_policy.h
index 99cb044..702c689 100644
--- a/include/hardware/audio_policy.h
+++ b/include/hardware/audio_policy.h
@@ -246,6 +246,9 @@ struct audio_policy {
     /* check if offload is possible for given sample rate, bitrate, duration, ... */
     bool (*is_offload_supported)(const struct audio_policy *pol,
                                 const audio_offload_info_t *info);
+
+    int (*set_parameters)(struct audio_policy *pol, const char *keyValuePairs);
+
 };
 
 
diff --git a/modules/audio/audio_policy.c b/modules/audio/audio_policy.c
index 9335654..aa750e1 100644
--- a/modules/audio/audio_policy.c
+++ b/modules/audio/audio_policy.c
@@ -236,6 +236,11 @@ static bool ap_is_offload_supported(const struct audio_policy *pol,
     return false;
 }
 
+static int ap_set_parameters(struct audio_policy *pol, const char *keyValuePairs)
+{
+    return 0;
+}
+
 static int create_default_ap(const struct audio_policy_device *device,
                              struct audio_policy_service_ops *aps_ops,
                              void *service,
@@ -287,6 +292,8 @@ static int create_default_ap(const struct audio_policy_device *device,
 
     dap->policy.is_offload_supported = ap_is_offload_supported;
 
+    dap->policy.set_parameters = ap_set_parameters;
+
     dap->service = service;
     dap->aps_ops = aps_ops;
 
-- 
1.9.2

